data_model/src/lib.rs

```diff
@@ -1804,6 +1810,89 @@ pub mod domain {
         }
     }
 
+    /// Represents path in IPFS. Performs some checks to ensure path validity.
+    ///
+    /// Should be constructed with `from_str()` method.
+    #[derive(
+        Debug,
+        Clone,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        Hash,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+    )]
+    pub struct IpfsPath(String);
+
+    impl FromStr for IpfsPath {
+        type Err = ParseError;
+
+        fn from_str(string: &str) -> Result<Self, Self::Err> {
+            let mut subpath = string.split('/');
+            let empty = subpath.next().ok_or(ParseError {
+                reason: "Impossible error",
```

Comment:
This needs a bit more explanation. Imagine that you're registering an account and get "Impossible error" in a red text-box.

Reply:
Yeah, but `split().next()` always gives first value. I'd use `unwrap()` or `expect()`, but clippy prevents it.
I'll add more info